### ISSUE TITLE
Default GitHub scopes are invasive

### ISSUE THREAD

**Issue by nicorikken** [ansible/galaxy-issues#159]
The default authorization of Ansible Galaxy for GitHub integration (which is mandatory) is quite invasive. Not yet having deployed my own role, but going from the [documentation](https://galaxy.ansible.com/intro) it seems that there is no actual use for write-access.

The current request:
![screen shot 2016-05-24 at 13 47 57](https://cloud.githubusercontent.com/assets/7458098/15502966/6658f332-21b7-11e6-9827-91c7c934dcb5.png)

Requesting such extensive [scopes](https://developer.github.com/v3/oauth/#scopes), without a clear reason upsets security people, preventing them from publishing to Ansible Galaxy.
_This issue directly originates from the remark in https://github.com/CISOfy/lynis-ansible/issues/2_

**Comment by mmckinst** [comment]
I've run in to this issue too, I don't mind giving ansible galaxy **READ** permissions to all public repos, but there's no reason for it to have write access.

**Comment by chouseknecht** [comment]
The scopes are defined [here](https://developer.github.com/v3/oauth/#scopes). We ask for 'public_repo' so that you can follow and star repositories from within Galaxy.

**Comment by leoarnold** [comment]
@chouseknecht I understand that this was the only way for you to implement GitHub starring in the Galaxy. I'd just like to emphasize that this clearly keeps me from publishing roles in Galaxy, because it is a major security hazard to give Ansible the ability to impersonate me on GitHub and possibly ruin all my reputation publicly and globally. Not that I suspect Ansible of intending this, but what when you get hacked or sold off to some fishy investor?

**Comment by mmckinst** [comment]
@leoarnold they've updated the scope to below which seems acceptable to me:

![screenshot from 2017-02-10 14-46-04](https://cloud.githubusercontent.com/assets/1631365/22847116/ed84aed0-ef9f-11e6-8011-41e5e9bc6881.png)

**Comment by leoarnold** [comment]
@mmckinst This is what I get when I try to sign in to the Galaxy at time of this writing and I'm really reluctant to grant "**read and write all public repository data**".
![screenshot_20170211_001454](https://cloud.githubusercontent.com/assets/4201235/22847772/9c7da000-efef-11e6-93d5-3851d2d704c7.png)

**Comment by mmckinst** [comment]
Weird. I just took a screenshot from https://github.com/settings/applications on my account but I gave galaxy permission for that stuff a long time ago. Back in August 2016 it was requesting more access when I tried to log in which is I how found this bug but at some point between then and now it stopped asking for all the extras perms when I logged in. Maybe there's a difference between new and old accounts? Hopefully someone from ansible will address this.
